keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c

```diff
@@ -0,0 +1,91 @@
+    // it'll be back to normal when you plug it back in.
+
+    [QWERTY] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_PSCR,          KC_MUTE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_DEL,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, MO(FNLAYER), KC_LALT,                            KC_SPC,                       KC_RALT, MO(FNLAYER),KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [FNLAYER] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          KC_MEDIA_PLAY_PAUSE,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET  ,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, RGB_VAI, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, RGB_VAD, _______
+    ),
+};
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if(IS_LAYER_ON(FNLAYER)) {
+     if (clockwise) {
+       tap_code(KC_MEDIA_NEXT_TRACK);
+     } else {
+       tap_code(KC_MEDIA_PREV_TRACK);
+     }
+  } else {
+     if (clockwise) {
+       tap_code(KC_VOLU);
+     } else {
+       tap_code(KC_VOLD);
+     }
+  }
+  return false;
+}
+
+static uint8_t val;
+
+void keyboard_post_init_user(void) {
+
+  val = rgb_matrix_get_val();
+}
+
+void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
+
+  if (host_keyboard_led_state().caps_lock) {
+    rgb_matrix_set_color_all(val,0,0);
+  } else {
+    rgb_matrix_set_color_all(val, val, val);
+    val = rgb_matrix_get_val();
+  }
+}
+
+void suspend_power_down_user(void) {
+
+  // rgb_matrix_set_color_all(0,0,0);
+  rgb_matrix_set_suspend_state(true);
+}
+
+void suspend_wakeup_init_user (void) {
+
+  // rgb_matrix_indicators_advanced_user(0, 0);
+  rgb_matrix_set_suspend_state(false);
+}
```

Comment:
```suggestion
```
Shouldn't be necessary anymore.

Reply:
OK removed although I seemed to need it. I must have been unlucky when testing if you're sure of it.

Reply:
Ah, maybe it is not necessary on develop, but that will be merging into master soon (and as a side note, you've opened this PR from your master branch).